### DISCUSSION TITLE
Javascript hook when component property changed

### DISCUSSION THREAD

**Discussion by jochemgruter** [livewire/livewire#3933] [1. Help] [answered]
Is there a javascript hook that will be fired when the value of a property in a component is changed? I can't find it in the documentations.

**Comment by jonneroelofs** [comment]
https://laravel-livewire.com/docs/2.x/lifecycle-hooks
If you scroll down there is a section Javascript hooks. Don't use them much so I am not sure if the cover your use case.

An alternative would be to entangle the property you want to monitor and use alpines watch directive.

**Comment by TVBZ** [comment]
> Is there a javascript hook that will be fired when the value of a property in a component is changed? I can't find it in the documentations.

Did  you find a solution? I am trying to get a computed property in JS but no such luck...
I have a basis computed property that returns filtered data. It works great in blade component, but in JS the data is not updated..

```
document.addEventListener("DOMContentLoaded", () => {
    Livewire.hook('message.processed', (message, component) => {
        console.log('message.processed');
        console.log({{Js::from($data)}}); // logs old data to console
    });
});
```

**Reply by joshhanley** [reply]
@TVBZ Livewire's frontend doesn't have access to the computed properties, so the best solution would probably be to save the data to a public property and do a check on that.

Hope this helps!

**Reply by TVBZ** [reply]
Yes that works. Thanks! :)

**Reply by jochemgruter** [reply]
Yes. I dit it this way too.

**Reply by joshhanley** [reply]
Ok great! 🙂

**Comment by joshhanley** [comment, marked as answer]
Copying answer here from comment above so it can be marked as answer:

Livewire's frontend doesn't have access to the computed properties, so the best solution would probably be to save the data to a public property and do a check on that.
